proc-macros/src/lib.rs

```diff
@@ -89,255 +57,21 @@ pub fn rpc_api(input_token_stream: TokenStream) -> TokenStream {
-					}
-				));
-			} else {
-				function_blocks.push(quote_spanned!(function.signature.span()=>
-					if request_outcome.is_none() && method == #rpc_method_name {
-						let request = server.request_by_id(&request_id).unwrap();
-						#(#params_builders)*
-						request_outcome = Some(Tmp::#variant_name { #(#params_names_list),* });
-					}
-				));
-
-				tmp_to_rq.push(quote_spanned!(function.signature.span()=>
-					Some(Tmp::#variant_name { #(#params_names_list),* }) => {
-						let request = server.request_by_id(&request_id).unwrap();
-						let respond = jsonrpsee::raw::server::TypedResponder::from(request);
-						return Ok(#enum_name::#variant_name { respond #(, #params_names_list)* });
-					},
-				));
-			}
-		}
-
-		let params_tys = generic_params.types.iter();
-
-		let tmp_generics = if generic_params.types.is_empty() {
-			quote!()
-		} else {
-			quote_spanned!(api.name.span()=>
-				<#(#params_tys,)*>
-			)
-		};
-
-		let on_request = quote_spanned!(api.name.span()=> {
-			#[allow(unused)]    // The enum might be empty
-			enum Tmp #tmp_generics {
-				#(#tmp_variants,)*
-			}
-
-			let request_id = r.id();
-			let method = r.method().to_owned();
-
-			let mut request_outcome: Option<Tmp #tmp_generics> = None;
-
-			#(#function_blocks)*
-
-			match request_outcome {
-				#(#tmp_to_rq)*
-				None => server.request_by_id(&request_id).unwrap().respond(Err(jsonrpsee::common::Error::method_not_found())),
-			}
-		});
-
-		let on_notification = quote_spanned!(api.name.span()=> {
-			let method = n.method().to_owned();
-			#(#notifications_blocks)*
-			// TODO: we received an unknown notification; log this?
-		});
-
-		let params_tys = generic_params.types.iter();
-
-		quote_spanned!(api.name.span()=>
-			#visibility async fn next_request(server: &'a mut jsonrpsee::raw::RawServer<R, I>) -> core::result::Result<#enum_name #ty_generics, std::io::Error>
-				where
-					R: jsonrpsee::transport::TransportServer<RequestId = I>,
-					I: Clone + PartialEq + Eq + std::hash::Hash + Send + Sync
-					#(, #params_tys: jsonrpsee::common::DeserializeOwned)*
-			{
-				loop {
-					match server.next_event().await {
-						jsonrpsee::raw::RawServerEvent::Notification(n) => #on_notification,
-						jsonrpsee::raw::RawServerEvent::SubscriptionsClosed(_) => unimplemented!(),       // TODO:
-						jsonrpsee::raw::RawServerEvent::SubscriptionsReady(_) => unimplemented!(),       // TODO:
-						jsonrpsee::raw::RawServerEvent::Request(r) => #on_request,
-					}
-				}
-			}
-		)
-	};
 
 	let client_impl_block = build_client_impl(&api)?;
-	let debug_variants = build_debug_variants(&api)?;
+	//let debug_variants = build_debug_variants(&api)?;
 
 	Ok(quote_spanned!(api.name.span()=>
-		#visibility enum #enum_name #tweaked_generics {
-			#(#variants),*
-		}
-
-		impl #impl_generics #enum_name #ty_generics #where_clause {
-			#next_request
-		}
+		#visibility struct #enum_name #tweaked_generics;
```

Comment:
this won't work for generics,

we would want to something similar to 

```rust
#visibility struct #enum_name #tweaked_generics(core::marker::PhantomData<(R, T, .... N)>);
```

`R, T, ... N` is in `#tweaked_generics` but is `<R, T>` so ideally we would convert it ti `<(R, N)>`

Reply:
Might be easier to just use the old code to use an enum where each type param is bound to a specific variant.